pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,284 @@
+	}
+	ec, err = encryptionconfiguration.LoadFromYAML(secretData)
+	if err != nil {
+		return nil, false, fmt.Errorf("EncryptionConfiguration in seed cluster (%v) is not consistent: %v", common.EtcdEncryptionSecretName, err)
+	}
+	_, ok = secretObj.Annotations[common.EtcdEncryptionForcePlaintextAnnotationName]
+	return ec, ok, nil
+}
+
+// readEncryptionConfigurationFromGarden reads the EncryptionConfiguration from the shoot namespace in the seed
+func (b *Botanist) readEncryptionConfigurationFromGarden(ctx context.Context) (*apiserverconfigv1.EncryptionConfiguration, error) {
+	secretNameInGardenCluster := generateGardenSecretName(b.Shoot.Info.Name, common.EtcdEncryptionSecretName)
+	secretObj := &corev1.Secret{}
+	err := b.K8sGardenClient.Client().Get(ctx, kutil.Key(b.Shoot.Info.Namespace, secretNameInGardenCluster), secretObj)
+	if err != nil {
+		return nil, err
+	}
+	secretData, ok := secretObj.Data[common.EtcdEncryptionSecretFileName]
+	if !ok {
+		return nil, fmt.Errorf("EncryptionConfiguration in garden cluster (%v) does not contain expected element: %v", secretNameInGardenCluster, common.EtcdEncryptionSecretFileName)
+	}
+	ec, err := encryptionconfiguration.LoadFromYAML(secretData)
+	if err != nil {
+		return nil, fmt.Errorf("EncryptionConfiguration in garden cluster (%v) is not consistent: %v", secretNameInGardenCluster, err)
+	}
+	return ec, nil
+}
+
+// updateEtcdEncryptionConfiguration updates the EncryptionConfiguration and computes the current configuration's checksum
+func (b *Botanist) updateEtcdEncryptionConfiguration(ctx context.Context, ecSeed *apiserverconfigv1.EncryptionConfiguration, forcePlaintextSecrets bool, needToWriteConfig bool) error {
+	ecSeedYamlBytes, err := encryptionconfiguration.ConvertToYAML(ecSeed)
+	if err != nil {
+		return err
+	}
+	if needToWriteConfig {
+		if err := b.writeEncryptionConfiguration(ctx, ecSeedYamlBytes, forcePlaintextSecrets); err != nil {
+			return err
+		}
+	}
+	b.mutex.Lock()
+	checksum := utils.ComputeSHA256Hex(ecSeedYamlBytes)
+	b.CheckSums[common.EtcdEncryptionSecretName] = checksum
+	b.Logger.Debugf("EncryptionConfiguration checksum: %v", checksum)
+	b.mutex.Unlock()
+	return nil
+}
+
+// writeEncryptionConfiguration writes the secret which contains the EncryptionConfiguration to the
+// shoot namespace in the seed cluster as well as to the garden cluster
+func (b *Botanist) writeEncryptionConfiguration(ctx context.Context, ecYamlBytes []byte, forcePlaintextSecrets bool) error {
+	//read old config from garden
+	secretNameInGardenCluster := generateGardenSecretName(b.Shoot.Info.Name, common.EtcdEncryptionSecretName)
+	oldGardenEC := &corev1.Secret{}
+	replaceGardenConfigurationOnSeedWriteError := true
+	if err := b.K8sGardenClient.Client().Get(ctx, kutil.Key(b.Shoot.Info.Namespace, secretNameInGardenCluster), oldGardenEC); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		replaceGardenConfigurationOnSeedWriteError = false
```

Comment:
I think we cannot continue here - You don't know which error exactly happened here, so I'd recommend to only replace  the secret on an exactly known error or just returning the error here and retrying later on.

Reply:
Hmm, I only continue if the error was a not found error. This can happen normally e.g. if we were to create the very first EncryptionConfiguration ... then there is none in Garden and this is fine.